packages/angular/src/generators/init-mfe/lib/get-remotes-with-ports.ts

```diff
@@ -0,0 +1,23 @@
+import type { Tree } from '@nrwl/devkit';
+import type { Schema } from '../schema';
+
+import { readProjectConfiguration } from '@nrwl/devkit';
+
+export function getRemotesWithPorts(host: Tree, options: Schema) {
+  // If type is shell and remotes supplied, check remotes exist
+  const remotesWithPort: { remoteName: string; port: number }[] = [];
+  if (
+    options.mfeType === 'shell' &&
+    Array.isArray(options.remotes) &&
+    options.remotes.length > 0
+  ) {
+    for (const remote of options.remotes) {
+      const remoteConfig = readProjectConfiguration(host, remote);
+      remotesWithPort.push({
+        remoteName: remote,
+        port: remoteConfig.targets['mfe-serve']?.options.port ?? 4200,
```

Comment:
If there are multiple remotes without the `port` option, should we generate incremental port numbers to avoid collisions? Or should we at least notify the dev that a port needs to be supplied?

Reply:
No, we'll just stick to `4200`. This is the default that would be expected when using the Angular application schematic. Anything we attempt to do could cause a collision. For example, a user might already have a remote set up at port `4201` so if we try to give port `4201` to a remote that we have found that doesn't have a `port` then we've caused a collision by trying to be smart. I think it's better if the onus is on the developer to ensure their configuration is correct.

With Module Federation, the developer inevitably has to touch config. We just want to reduce some of the pain of scaffolding that.